File: internal/content/testdata/bench/rust/nc-best-time-cooldown.rs
fn max_profit(prices: Vec<i64>) -> i64 {
    let neg_inf = i64::MIN / 2;
    let mut hold = neg_inf;
    let mut sold = 0i64;
    let mut rest = 0i64;
    for &price in prices.iter() {
        let prev_sold = sold;
        sold = hold + price;
        hold = hold.max(rest - price);
        rest = rest.max(prev_sold);
    }
    sold.max(rest)
}

File: internal/content/testdata/bench/go/nc-best-time-cooldown.go
func max_profit(prices []int) int {
	hold := -1 << 62
	sold := 0
	rest := 0
	for _, price := range prices {
		prevSold := sold
		sold = hold + price
		if rest-price > hold {
			hold = rest - price
		}
		if prevSold > rest {
			rest = prevSold
		}
	}
	if sold > rest {
		return sold
	}
	return rest
}
